src/components/DayPickerRangeController.jsx

```diff
@@ -558,14 +564,14 @@ export default class DayPickerRangeController extends React.PureComponent {
         const start = getSelectedDateOffset(startDateOffset, day);
         const end = getSelectedDateOffset(endDateOffset, day, rangeDay => rangeDay.add(1, 'day'));
 
-        dateOffset = {
+        nextDateOffset = {
           start,
           end,
         };
 
         // eslint-disable-next-line react/destructuring-assignment
-        if (this.state.dateOffset && this.state.dateOffset.start && this.state.dateOffset.end) {
-          modifiers = this.deleteModifierFromRange(modifiers, this.state.dateOffset.start, this.state.dateOffset.end, 'hovered-offset');
+        if (this.state.nextDateOffset && dateOffset.start && dateOffset.end) {
```

Comment:
@monokrome I don't think `this.state.nextDateOffset` exists. Did you mean just `nextDateOffset`? Also, I think that it is always truthy so this check doesn't really make sense. 

I think that this is checking for the existing state and so should just use the deconstructed `dateOffset` and this line should be:
```
if (dateOffset && dateOffset.start && dateOffset.end) {
```

Reply:
@majapw Yep! Seems right. This was using `this.state`directly instead of the desctructured value, which seems like this is only for an edge-case that wasn't represented by tests. 🙀 

Problem solved now! Thanks for finding it :)